Rename PREFIX to DATADIR as its only purpose is to access data files
(and avoid conflict with FreeBSD PREFIX, which has another meaning)

Tune GAMELIBS for FreeBSD platform

--- Makefile.orig	2020-09-22 22:08:35 UTC
+++ Makefile
@@ -14,7 +14,7 @@
 RELEASE ?= 1
 
 # Base path of app installation
-PREFIX ?= /usr/local/share/games/jfsw
+DATADIR ?= /usr/local/share/games/jfsw
 
 # Engine source code path
 EROOT ?= jfbuild
@@ -168,6 +168,10 @@ ifeq ($(PLATFORM),LINUX)
 	NASMFLAGS+= -f elf
 	GAMELIBS+= $(JFAUDIOLIB_LDFLAGS)
 endif
+ifeq ($(PLATFORM),BSD)
+	NASMFLAGS+= -f elf
+	GAMELIBS+= $(JFAUDIOLIB_LDFLAGS) -pthread
+endif
 ifeq ($(PLATFORM),WINDOWS)
 	OURCFLAGS+= -I$(DXROOT)/include
 	NASMFLAGS+= -f win32 --prefix _
@@ -207,7 +211,7 @@ OURCFLAGS+= $(BUILDCFLAGS)
 LIBS+= $(BUILDLIBS)
 
 ifneq ($(PLATFORM),WINDOWS)
-	OURCFLAGS+= -DPREFIX=\"$(PREFIX)\"
+	OURCFLAGS+= -DDATADIR=\"$(DATADIR)\"
 endif
 
 .PHONY: clean all engine $(ELIB)/$(ENGINELIB) $(ELIB)/$(EDITORLIB) $(AUDIOLIBROOT)/$(JFAUDIOLIB)
